src/drawing/canvas.rs

```diff
@@ -0,0 +1,65 @@
+use image::{
+    GenericImage,
+    GenericImageView,
+    Pixel
+};
+
+/// A surface for drawing on - many drawing functions in this
+/// library are generic over a `Canvas` to allow the user to
+/// configure e.g. whether to use blending.
+///
+/// All instances of `GenericImage` implement `Canvas`, with
+/// the behaviour of `draw_pixel` being equivalent to calling
+/// `set_pixel` with the same arguments.
```

Comment:
This could use an example of using a `GenericImage` to show that it really is drop-in. 

It would be good to have a mention of `Blend` in the documentation as well, as motiviation for this intermediate trait and as an implementation guide for other `Canvas` customizations.

Reply:
👍 Done